faucet/models.py

```diff
@@ -421,3 +424,29 @@ class DonationReceipt(models.Model):
 
     class Meta:
         unique_together = ("faucet", "tx_hash")
+
+
+class DonationContract(SafeDeleteModel):
+    contract_address = models.CharField(max_length=255, blank=False, null=False)
+    faucet = models.ForeignKey(
+        Faucet,
+        related_name="donation_contracts",
+        on_delete=models.PROTECT,
+        null=False,
+        blank=False,
+    )
```

Comment:
I think it's better to have a status (active/inactive) field, so we're able to deactivate an old donation contract and define new active one for the faucet.

Reply:
I use safedelete so, we have similar functionality.